include/ace/managers/mouse.h

```diff
@@ -1,101 +1,222 @@
 #ifndef GUARD_ACE_MANAGER_MOUSE_H
 #define GUARD_ACE_MANAGER_MOUSE_H
 
-#ifdef AMIGA
-#include <clib/exec_protos.h> // Amiga typedefs
-#include <clib/intuition_protos.h> // IDCMP_RAWKEY etc
-#include <devices/input.h>
-#include <clib/alib_protos.h>
-#endif // AMIGA
-
 #include <ace/types.h>
+#include <ace/macros.h>
 
-#include <ace/managers/window.h>
-
-/* Types */
-#ifdef AMIGA
-#define MOUSE_LMB IECODE_LBUTTON
-#define MOUSE_RMB IECODE_RBUTTON
-#define MOUSE_MMB IECODE_MBUTTON
-#else
-#define MOUSE_LMB 1
-#define MOUSE_RMB 2
-#define MOUSE_MMB 4
-#endif // AMIGA
+// Mouse button identifiers
+#define MOUSE_LMB 0
+#define MOUSE_RMB 1
+#define MOUSE_MMB 2
 
+// Button state defines
 #define MOUSE_NACTIVE 0
 #define MOUSE_USED 1
 #define MOUSE_ACTIVE 2
 
-typedef struct {
-	UBYTE pStates[3];
+// Mouse ports
+// #define MOUSE_PORT_0 0 // Unused mouse port for alignment
+#define MOUSE_PORT_1 1
+#define MOUSE_PORT_2 2
+
+/* Types */
+typedef struct _tMouse {
+	UWORD uwX;
+	UWORD uwY;
+	UBYTE pButtonStates[3];
+#ifdef AMIGA
+	UBYTE ubPrevHwX;
+	UBYTE ubPrevHwY;
+#endif
+} tMouse;
+
+typedef struct _tMouseManager {
+	UBYTE ubPortFlags;
+	tMouse pMouses[3]; ///< Zero is pad, faster than subtracting from port code.
+	UWORD uwLoX; ///< Minimum cursor's X position
+	UWORD uwLoY; ///< Minimum cursor's Y position
+	UWORD uwHiX; ///< Maximum cursor's X position
+	UWORD uwHiY; ///< Maximum cursor's Y position
 #ifdef AMIGA
-	__chip UWORD pBlankCursor[6];
-	struct MsgPort *pInputMP;
-	struct IOStdReq *pInputIO;
+	UWORD uwPrevPotGo; ///< Previous control port config.
 #endif // AMIGA
 } tMouseManager;
 
 /* Globals */
 extern tMouseManager g_sMouseManager;
 
 /* Functions */
-void mouseOpen(void);
 
-void mouseSetState(
-	IN UBYTE ubMouseCode,
-	IN UBYTE ubMouseState
+/**
+ * Initializes mouse management for given ports.
+ * @param ubPortFlags: Ports in which mouse should be processed.
+ *        OR combination of MOUSE_PORT_* defines.
+ * @see mouseDestroy()
+ * @see mouseProcess()
+ */
+void mouseCreate(
+	IN UBYTE ubPortFlags
 );
 
-UBYTE mouseCheck(
-	IN UBYTE ubMouseCode
-);
+/**
+ * Cleans up after mouse maanger.
+ * @see mouseCreate()
+ */
+void mouseDestroy(void);
 
-UBYTE mouseUse(
-	IN UBYTE ubMouseCode
-);
+/**
+ * Processes mouse manager, updating mice's position and button states.
+ * Should be called once per frame.
+ */
+void mouseProcess(void);
 
-UBYTE mouseIsIntersects(
-	IN UWORD uwX,
-	IN UWORD uwY,
-	IN UWORD uwWidth,
-	IN UWORD uwHeight
+/**
+ * Set on-screen constraints for cursor.
+ * @param uwLoX Minimum cursor X position.
+ * @param uwLoX Minimum cursor Y position.
+ * @param uwHiY Maximum cursor X position.
+ * @param uwHiY Maximum cursor Y position.
+ */
+void mouseSetBounds(
```

Comment:
When we have support for "multiple" simultaneous mouse handling... We should be able to set bounds for specific mouse.

Think about possibility to make split screen games with 2 connected mouses.

Reply:
Done in 776fae8